sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FilterPushdownSuite.scala

```diff
@@ -1153,12 +1154,35 @@ class FilterPushdownSuite extends PlanTest {
       "x.a".attr === Rand(10) && "y.b".attr === 5))
     val correctAnswer =
       x.where("x.a".attr === 5).join(y.where("y.a".attr === 5 && "y.b".attr === 5),
-        condition = Some("x.a".attr === Rand(10)))
+        joinType = Cross).where("x.a".attr === Rand(10))
```

Comment:
this is not a change we want, right?

Reply:
As the code in canEvaluateWithinJoin, we can get the scope relation : (CannotEvaluateWithinJoin = nonDeterminstic +  Unevaluable) > Unevaluable > PythonUDF. 
So for the safety maybe I just limit the change scope to the smallest PythonUDF only. Need some advise from you thanks :)


https://github.com/apache/spark/blob/0736e72a66735664b191fc363f54e3c522697dba/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala#L104-L120

Reply:
cc @cloud-fan @gatorsmile for advice on this. It may probably be ok, as it lets supporting a case which was not supported before. But I am not sure about the added value as performing a cross join is often an impossible operation.

Reply:
Thanks @mgaido91 for the detailed review and advise, for me, I maybe choose only limited the change scope to pythonUDF only or at lease Unevaluable only. Waiting for others advice.